java/lang/Float/Float.cpp

```diff
@@ -142,68 +140,483 @@ Float Float::operator/=(const Float &target) const {
     return (Float) (this->original / target.original);
 }
 
-char Float::charValue() const {
-    string convertResult = string_from_float(this->original);
-    char charValueResult = string_to_char(convertResult);
-    free(convertResult);
-    return charValueResult;
-}
-// TODO(thoangminh): enable after finish toString()
-//string Float::stringValue() const {
-//    return (string) this->toString();
-//}
-
 float Float::floatValue() const {
     return this->original;
 }
 
 short Float::shortValue() const {
-    return (short) this->original;
+    return static_cast<short> (this->original);
 }
 
 int Float::intValue() const {
-    return (int) this->original;
+    return static_cast<int> (this->original);
 }
 
-/**
- * Long value
- *
- * @return long
- */
 long Float::longValue() const {
-    return (long) this->original;
+    return static_cast<long> (this->original);
 }
 
-/**
- * Float value
- *
- * @return
- */
 double Float::doubleValue() const {
-    return this->original;
+    return static_cast<double> (this->original);
+}
+
+byte Float::byteValue() const {
+    return static_cast<byte> (this->original);
 }
 
-/**
- * Float to String
- *
- * @return String
- */
 string Float::toString() const {
     return this->originalString;
 }
 
-/**
- * String to Float
- *
- * @param target
- * @return Float
- */
+Float Float::parseFloat(String inputString) {
+    return static_cast<Float> (string_to_float(inputString.toString()));
+}
+
+boolean Float::isFinite(float valueFloat) {
+    return (Math::abs(valueFloat) <= MAX_VALUE);
+}
+
+boolean Float::isInfinite(float valueFloat) {
+    boolean isPOSITIVE_INFINITY = (valueFloat == POSITIVE_INFINITY);
+    boolean isNEGATIVE_INFINITY = (valueFloat == NEGATIVE_INFINITY);
+    return (isPOSITIVE_INFINITY || isNEGATIVE_INFINITY) ;
+}
+
+boolean Float::isInfinite() {
+    return isInfinite(this->original);
+}
+
+boolean Float::isNaN(float valueFloat) {
+    return (valueFloat != valueFloat);
+}
+
+boolean Float::isNaN() {
+    return isNaN(this->original);
+}
+
+Float Float::valueOf(String stringInput) {
+    return Float::parseFloat(stringInput);
+}
+
+Float Float::valueOf(float inputFloat) {
+    return Float(inputFloat);
+}
+
+String Float::floatToBinary32StringType(float floatInput) {
+    string integerPartNormalizeForm = (string) calloc (280, sizeof(char));
+    string fractionPartNormalizeForm = (string) calloc (25, sizeof(char));
+    string floatInputNormalizeForm = (string) calloc (280, sizeof(char));
+    string resultFloatToBinary32StringType
+            = (string) calloc (33, sizeof(char));
+
+    int powerExponentBase2 = 0;
+    int integerPartFloatInput;
+
+    int sizeOfIntegerPartNormalizeForm;
+    int sizeOfFractionPartNormalizeForm;
+    int sizeOfFloatInputNormalizeForm;
+
+    int sizeOfExponentPartBinary32 = 8;
+
+    int i;

```

Comment:
Please rename this variable and others variable name "i"

Reply:
Thanks. Fixed.